authors/apps/core/password_reset_manager.py

```diff
@@ -0,0 +1,75 @@
+from ..authentication.models import UserManager, User
+from rest_framework import serializers
+import jwt, datetime
+from django.conf import settings
+from django.template.loader import render_to_string
+from django.core.mail import send_mail
+
+class PasswordResetManager:
+    """
+    This class handles the operations involved in the email-based
+    password-reset feature including preparing the email message,
+    reset link and verification of link and updating password
+    """
+
+    def __init__(self, request):
+        self.sender_email = "teamabasama@gmail.com"
+        self.subject = "You requested a password reset"
+        self.password_reset_url = request.build_absolute_uri('/api/v1/users/reset-password/')
+
+    def get_user_by_email(self, email):
```

Comment:
@pbkabali  Could you please add descriptive docstring messages below the methods as well

Reply:
@missvicki Working on that. Thanks  a lot